src/log4net/Util/ThreadContextStack.cs

```diff
@@ -188,10 +188,10 @@ internal Stack InternalStack
     set
     {
       _stack.Clear();
-      var frames = (StackFrame[])value.ToArray();
+      object[] frames = value.ToArray();
       for (int i = frames.Length - 1; i >= 0; i--)
       {
-        _stack.Push(frames[i]);
+        _stack.Push(frames[i].EnsureIs<StackFrame>());
```

Comment:
minor: I'd suggest rather using pettern-matching instead of throwing if something in that array is wrong, eg

```csharp
if (frames[i] is StackFrame s)
{
  _stack.Push(s)
}
```
This would stop a throw on invalid data set, but leave the collection empty, which may not be better - this may just be a stylistic choice on my part.